backend/pkg/redis/redis.go

```diff
@@ -155,3 +166,44 @@ func ClientFromConnectionString(
 		Result()
 	return rdb, err
 }
+
+// Invalidates a cache entry by swapping it with a temporary short-lived entry.
+// When a function receives this temporary value, it should count it as a cache miss,
+// but also stop the caching of the value fetched DB value.
+func SafeCacheDel(ctx context.Context, c Client, key string) error {
```

Comment:
Invalidating and deleting are two separate operations.
```suggestion
func InvalidateCache(ctx context.Context, c Client, key string) error {
```